third_party/amd/lib/TritonAMDGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -209,9 +210,17 @@ struct ConvertTritonAMDGPUToLLVM
     mlir::arith::populateArithToLLVMConversionPatterns(typeConverter, patterns);
     mlir::populateMathToLLVMConversionPatterns(typeConverter, patterns);
 
+    std::string chipset = targetInfo.getArch();
```

Comment:
We don't need this as `this->arch` is already it? See L84. Also maybe good to move check to the beginning before L84.